### PR TITLE
Option to enable convert kwargs to snake case globally

### PR DESCRIPTION
**Pull request by enwawerueli**
A global option for the `convert_kwargs_to_snake_case` decorator would be nice to have. This is especially useful if this is the desired behavior for all resolvers

### CODE REVIEW

**Comment by rafalp** [comment]
Hi! There's better way to enable this behavior than through decorator. Every input field and argument has `out_name` attribute that can be set to custom value. Eg. if you set `out_name` on field `name` to `bob`, `name: "test"` in GraphQL query will be passed to `bob` kwarg/dict key. This is approach that Ariadne GraphQL Modules uses:

- Input fields: https://github.com/mirumee/ariadne-graphql-modules/blob/main/ariadne_graphql_modules/input_type.py#L117
- Fields args: https://github.com/mirumee/ariadne-graphql-modules/blob/main/ariadne_graphql_modules/object_type.py#L123

We could have an utility function that iterates schema types and if those types are `GraphQLInputType` or `GraphQLObjectType` we could set out names on those types. This util would be then called by `make_executable_schema` to do it's magic.

**Comment by enwawerueli** [comment]
Noted. I haven't tried out graphql-modules, so I'll have a closer look at that first

File: ariadne/objects.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def bind_resolvers_to_graphql_type(self, graphql_type, replace_existing=True):
                 )
             if graphql_type.fields[field].resolve is None or replace_existing:
                 graphql_type.fields[field].resolve = resolver
-
+
```
Comment (rafalp) [inline ariadne/objects.py:60]: Format code with black.

File: ariadne/executable_schema.py [hunk 2]
```diff
@@ -40,6 +43,9 @@ def make_executable_schema(
     validate_schema_enum_values(schema)
     repair_default_enum_values(schema, flat_bindables)
 
+    if kwargs_to_snake_case:
```
Comment (rafalp) [inline ariadne/executable_schema.py:46]: I'm curious if this should be an option or just a utility that you can run against your schema separately. But I guess the latter would require us to create a utility that copies whole schema so we don't mutate it.

File: ariadne/executable_schema.py [hunk 3]
```diff
@@ -14,12 +15,14 @@
...
 from .schema_visitor import SchemaDirectiveVisitor
 from .types import SchemaBindable
+from .utils import convert_camel_case_to_snake
 
 
 def make_executable_schema(
     type_defs: Union[str, List[str]],
     *bindables: Union[SchemaBindable, List[SchemaBindable]],
     directives: Dict[str, Type[SchemaDirectiveVisitor]] = None,
+    kwargs_to_snake_case: bool = False
```
Comment (rafalp) [inline ariadne/executable_schema.py:25]: How about `convert_args_to_snake_case`?

File: ariadne/executable_schema.py [hunk 4]
```diff
@@ -65,3 +71,15 @@ def repair_default_enum_values(schema, bindables) -> None:
...
         if isinstance(bindable, EnumType):
             bindable.bind_to_default_values(schema)
+
+
+def set_outname_to_snake_case(schema) -> None:
+    for graphql_type in schema.type_map.values():
+        if not isinstance(graphql_type, (GraphQLObjectType, GraphQLInputObjectType)):
+            continue
+        for field_name, field in graphql_type.fields.items():
+            field.out_name = convert_camel_case_to_snake(field_name)
```
Comment (rafalp) [inline ariadne/executable_schema.py:81]: We should do `if isinstance(graphql_type, GraphQLInputObjectType)` and then set field.out_name, then `if isinstance(graphql_type, GraphQLObjectType)` do that for args.

File: ariadne/executable_schema.py [hunk 5]
```diff
@@ -65,3 +71,15 @@ def repair_default_enum_values(schema, bindables) -> None:
     for bindable in bindables:
         if isinstance(bindable, EnumType):
             bindable.bind_to_default_values(schema)
+
+
+def set_outname_to_snake_case(schema) -> None:
+    for graphql_type in schema.type_map.values():
+        if not isinstance(graphql_type, (GraphQLObjectType, GraphQLInputObjectType)):
```
Comment (rafalp) [inline ariadne/executable_schema.py:78]: We should do `if isinstance(graphql_type, GraphQLInputObjectType)` and then set field.out_name, then `if isinstance(graphql_type, GraphQLObjectType)` do that for args.

**Review by rafalp — CHANGES_REQUESTED** [review]
Looking okay, but could you please add tests?

File: ariadne/executable_schema.py [hunk 6]
```diff
@@ -65,3 +71,15 @@ def repair_default_enum_values(schema, bindables) -> None:
     for bindable in bindables:
         if isinstance(bindable, EnumType):
             bindable.bind_to_default_values(schema)
+
+
+def set_outname_to_snake_case(schema) -> None:
```
Comment (rafalp) [inline ariadne/executable_schema.py:76]:
```suggestion
def set_outnames_to_snake_cases(schema) -> None:
```

**Comment by enwawerueli** [comment]
@rafalp I've incorporated the changes you suggested. Should be all good!

**Comment by enwawerueli** [comment]
Also noted a name like `complexField_Int32` converts to `complex_field__int_32`, not sure if that's the intended behavior or it should be `complex_field_int_32`. Not a problem though

File: ariadne/executable_schema.py [hunk 7]
```diff
@@ -65,3 +71,14 @@ def repair_default_enum_values(schema, bindables) -> None:
     for bindable in bindables:
         if isinstance(bindable, EnumType):
             bindable.bind_to_default_values(schema)
+
+
+def set_outnames_to_snake_case(schema) -> None:
```
Comment (rafalp) [inline ariadne/executable_schema.py:76]:
```suggestion
def set_out_names_to_snake_case(schema) -> None:
```

File: ariadne/executable_schema.py [hunk 8]
```diff
@@ -40,6 +43,9 @@ def make_executable_schema(
     validate_schema_enum_values(schema)
     repair_default_enum_values(schema, flat_bindables)
 
+    if convert_args_to_snake_case:
+        set_outnames_to_snake_case(schema)
```
Comment (rafalp) [inline ariadne/executable_schema.py:47]:
```suggestion
        set_out_names_to_snake_case(schema)
```

**Comment by rafalp** [comment]
Left some extra notes but those are tiny bits and nothing blocking - but code needs to be formatted with black to pass tests.

**Comment by rafalp** [comment]
Thank you for contributing this. Can you say if approach from #1007 also works for you?

Thank you!

**Pull request closed without merging**